src/TLuaInterpreter.cpp

```diff
@@ -2598,7 +2598,7 @@ int TLuaInterpreter::setExitStub(lua_State* L)
     }
     TRoom* pR = host.mpMap->mpRoomDB->getRoom(roomId);
     if (!pR) {
-        lua_pushstring(L, "setExitStub: RoomId doesn't exist");
```

Comment:
It seems strange that we are not reporting the value that `roomId` was given! Tthough if we fix this so it does, we should report the original argument passed as the first argument and NOT what `getVerifiedInt(...)`  returned.

Reply:
Because the "number" that the user entered might not exactly be an "integer" like wot we wuz expecting, or could contain extra things (like commas that are not parsed as the user might expect) e.g. `1,200` is handled not as `1200` but as `1` and `10.6` is handled as `10` not `10.6` or `11`.

Reply:
I don't think we do it like this anywhere else in the API... has this been a problem that has been affecting people a lot?

Reply:
I think I started doing it like that for one of my PRs recently, ah yeah, it was #6401 where I added `TLuaInterpreter::getVerifiedStringOrInteger(...)` which uses `lua_tonumber(...)` but then applies `qRound(...)` to convert the `double` the lua function returns into an integer (and avoid using `lua_tointeger(...)` because {IIRC} on Windows `lua_tointeger(...)` returns a `long int` that may not fit into an `int` on that platform which the built-in Qt code analyser moans about...!) As such, when then reporting an issue with the argument supplied it has to refer to the passed value rather than what it might have gotten mangled to...!

:grinning:

Reply:
Is this addressing a real problem people have been running into though?

Reply:
I think I will skip doing this change, I was just looking to do a simple fix and there is a lot of other higher priority things to be working on.